### PR TITLE
Adds staging directory for kubectl code

### PR DESCRIPTION
**Pull request by seans3**
* Creates staging directory for kubectl code
* Adds the following initial files to this directory:
  - .github/PULL_REQUEST_TEMPLATE.md
  - code-of-conduct.md
  - LICENSE
  - OWNERS
  - README.md
  - SECURITY_CONTACTS
* Code committed to the kubectl staging directory will be published to: https://github.com/kubernetes/kubectl

For more infomation please see the initial KEP describing kubectl staging move:
https://github.com/kubernetes/enhancements/pull/949

as well as:
https://github.com/kubernetes/enhancements/pull/1028

```release-note
NONE
```

/kind cleanup
/sig cli
/area kubectl

### CODE REVIEW

**Comment by liggitt** [comment]
needs publishing rules update (see verify failures)

is the kubectl repo cleared out and ready for this to publish to it?

**Comment by seans3** [comment]
@liggitt My current plan is:

1) create the kubectl staging directory with this PR.
2) clear out the kubectl repository (current in progress with https://github.com/kubernetes/kubectl/pull/655).
3) (only after completing the previous two steps) turn on the publishing bot.

Please let me know if this is a reasonable approach.

**Comment by liggitt** [comment]
> @liggitt My current plan is:
> 
> 1. create the kubectl staging directory with this PR.
> 2. clear out the kubectl repository (currently in progress with [kubernetes/kubectl#655](https://github.com/kubernetes/kubectl/pull/655)).
> 3. (only after completing the previous two steps) turn on the publishing bot.

the publishing bot is already running, so steps 1 and 2 need to be flipped

**Comment by seans3** [comment]
Hi @liggitt. The kubectl repo has been wiped. I believe this PR is suitable to be merged.

**Comment by lavalamp** [comment]
There are still a ton of commits over there, I am not sure how the publishing bot is going to react to that?

/approve
/lgtm
/hold

Please get the publishing bot owners to ack this before unholding.

File: staging/publishing/rules.yaml
Comment (nikhita) [inline staging/publishing/rules.yaml]: This needs to be release-12.0 since client-go has it's own release process.

File: staging/src/k8s.io/kubectl/SECURITY_CONTACTS
Comment (nikhita) [inline staging/src/k8s.io/kubectl/SECURITY_CONTACTS]: I think should be s/jessfraz/joelsmith. Ref: https://github.com/kubernetes/community/tree/master/committee-product-security#members

(probably needs update for all staging repos, but that can be done in another PR IMO)

cc @liggitt

File: staging/publishing/rules.yaml
Comment (nikhita) [inline staging/publishing/rules.yaml]: Since the repo is empty right now, do you want to add these dependencies when you start populating the repo?

Having them here won't harm but in case you want to keep them here, can you also add kubectl to https://github.com/kubernetes/kubernetes/blob/master/staging/publishing/import-restrictions.yaml? :)

**Comment by nikhita** [comment]
> There are still a ton of commits over there, I am not sure how the publishing bot is going to react to that?

TLDR: This will not hurt the publishing-bot. :+1: 

Given that there are no files in the repo, the publishing-bot will treat it as a new repo and start publishing normally. Ref: https://github.com/kubernetes/publishing-bot/blob/master/artifacts/scripts/util.sh#L151-L153

Once the comments are addressed, this PR should be good to be merged. The actual turning of the publishing-bot needs some more tweaks, detailed below.

---

I have created https://github.com/kubernetes/publishing-bot/pull/182 and https://github.com/kubernetes/test-infra/pull/12794 to help with turning on the publishing-bot.

- There were some teams/people who had direct write/admin access to the kubectl repo. Like we do for other staging repos, I've removed access to these teams/people and I've given admin access to the @kubernetes/stage-bots team.

- https://github.com/kubernetes/test-infra/pull/12794 will enable the appropriate branch protection setting for the repo.

- The publishing bot isn't running yet, it will only be turned on after https://github.com/kubernetes/publishing-bot/pull/182 has merged. After the test-infra PR gets merged _and_ this PR gets merged, we can merge the publishing-bot repo. This will trigger the actual publishing process.

**Comment by nikhita** [comment]
One small nit: can you also add the repo to https://github.com/kubernetes/kubernetes/tree/master/staging#external-repository-staging-area? :)

File: staging/src/k8s.io/kubectl/SECURITY_CONTACTS
Comment (nikhita) [reply staging/src/k8s.io/kubectl/SECURITY_CONTACTS]: > (probably needs update for all staging repos, but that can be done in another PR IMO)

created https://github.com/kubernetes/kubernetes/pull/78470

File: staging/publishing/rules.yaml
Comment (seans3) [reply staging/publishing/rules.yaml]: I added the allowed imports into `import-restrictions.yaml`

**Comment by nikhita** [comment]
@seans3 lgtm overall, but looks like the commit message needs update :)

**Comment by seans3** [comment]
> Sean Sullivan lgtm overall, but looks like the commit message needs update :)

Thanks for pointing that out. I made a mistake while squashing the commit :)

**Review by nikhita — APPROVED** [review]
:rocket: 

/lgtm
/hold cancel

**Comment by nikhita** [comment]
https://github.com/kubernetes/kubectl has been published! :tada:

**Pull request merged**
